File: source/js-codegen/src/main/java/org/tlsys/BBB.java
package org.tlsys;

/**
 * Created by subochev on 26.09.16.
 */
public class BBB {
    public BBB(int a, int c, int d) {
        AAAKt.testAAA(a, c, d);

        do {

            if (a > 999999) {
                AAAKt.testAAA(111, 111, 111);
            } else {
                AAAKt.testAAA(222, 222, 222);
            }

            a = a - 22222;
        } while (a > 999999);


        try {
            AAAKt.testAAA(333, 333, 333);
        } catch (Throwable e) {
            AAAKt.testAAA(444, 444, 444);
        }/* finally {
            AAAKt.testAAA(555, 555, 555);
        }*/
    }
}

File: source/js-codegen/src/main/java/org/tlsys/BBB.kt
package org.tlsys

import java.util.HashMap

class BBB(a: Int, c: Int, d: Int) {
    init {
        var b = a
        testAAA(b, c, d)
        F()
        if (b > 1000)
            testAAA(111, 111, 111)

/*
        if (b>1000) {
            if (b<=500) {
                testAAA(111, 111, 111)
            }
            else {
                testAAA(222, 222, 222)
            }
        } else {
            testAAA(333,333,333)
        }
        */
    }
}
